docs/pages/api.rst

```diff
@@ -0,0 +1,121 @@
+.. _nuts-discovery-api:
+
+***
+Api
+***
+
+The *Nuts Discovery Service* consists of 2 api's: the network map and certificare api's. The certificate api is in the initial setup phase of the Corda node. To publish the node details with an electronic signature and retrieve the signed certificate from the *Nuts Discovery Service*. The network map api is used to retrieve details about all the nodes that are connected to the network.
+
+Certificate API
+===============
+
+The *Certificate API* is part of the Corda node but it's seems to be lacking documentation. After reverse engineering, 2 requests could be identified. The initation process for a Corda node can be started with:
+::
+
+    java -jar corda.jar initial-registration -p changeit
+
+where the given password is the password of the root truststore. More info on how to start and configure a node can be found in :ref:`node setup <nuts-discovery-setup>`.
+
+When started, the Corda node will create a couple of keystore as seen `here <https://docs.corda.net/permissioning.html#certificate-hierarchy>`_. The node will send a certificate request to the configured *Doorman* url followed by /certificate. The return value will be an identifier which the node will use to poll the service. When the service has signed the request, the certificate is then downloaded and put in the keystore. Below are the details of the two services.
+
+.. http:post:: /certificate
+
+   Receiving endpoint for a certificate request (CSR). The body must be in PKCS10 byte format. The response is a plain text response with an identifier which can be used in the GET call.
+
+   :reqheader Platform-Version: the Corda platform version (4)
+   :reqheader Client-Version: todo
+   :reqheader Private-Network-Map: todo
+
+   :statuscode 200: no error, body is a plain text string identifier
+   :statuscode 400: wrong format certificate signing request
+
+.. http:get:: /(request_id)
+
+   Retrieve the signed certificate which was requesting using the POST method. `request_id` is the returned identifier from the POST request. The body contains a zip file with 3 files: **cordaclientca.cer**, **cordaintermediateca.cer** and **cordarootca.cer** (in this order). Each file is the ASN.1 DER encoding of a X.509 certificate.
+
+   :resheader Content-Disposition: (attachment; filename="certificates.zip")
+
+   :statuscode 200: Certificate request has been signed, the body consists of a zip file with a list of certificates.
+   :statuscode 400: something went wrong.
+   :statuscode 404: Unknown identifier or request hasn't been signed yet.
```

Comment:
So, this endpoint should be polled. If the request has not been signed you get a 404?
If the request has been signed, how often can the `request_id` be used?